tenant/resolver.go

```diff
@@ -3,18 +3,14 @@ package tenant
 import (
 	"context"
 	"errors"
-	"net/http"
 	"strings"
 
 	"github.com/grafana/dskit/user"
 )
 
-var defaultResolver Resolver = NewSingleResolver()
-
-// WithDefaultResolver updates the resolver used for the package methods.
-func WithDefaultResolver(r Resolver) {
-	defaultResolver = r
-}
+var (
+	errInvalidTenantID = errors.New("invalid tenant ID")
```

Comment:
can we add some details about what's invalid (`"tenant ID is ., .. or contains / or \"`)

Reply:
I'm realizing now that it doesn't make sense to have `containsUnsafePathSegments` separate from the logic in `ValidTenantID`. WDYT about moving the logic from `containsUnsafePathSegments` to `ValidTenantID`?

Reply:
Fixed in cb0052572fc643943975681cad008d9610787447 and moved that logic into `ValidTenantID()`